### PR TITLE
Custom drp op

### PR DESCRIPTION
**Pull request by mykhailoB8**

### CODE REVIEW

File: src/components/CalendarDay.jsx [hunk 1]
```diff
@@ -257,6 +263,34 @@ export default withStyles(({ reactDates: { color, font } }) => ({
       color: color.selectedSpan.color_active,
     },
   },
+  CalendarDay__selectedInvalid_span: {
+    background: '#FFF2F2',
```
Comment (tofaelahmed) [inline src/components/CalendarDay.jsx:267]: Can you use values from theme like how its being used in other styles?

File: src/components/CalendarMonth.jsx [hunk 2]
```diff
@@ -174,13 +212,40 @@ class CalendarMonth extends React.PureComponent {
...
     const { weeks } = this.state;
     const monthTitle = renderMonthText ? renderMonthText(month) : month.format(monthFormat);
 
+    const currentMonth = month.clone();
     const verticalScrollable = orientation === VERTICAL_SCROLLABLE;
 
+    let lastInvalidWeek = null;
+
+    weeks.forEach((week, i) => {
+      week.forEach((day) => {
```
Comment (tofaelahmed) [inline src/components/CalendarMonth.jsx]: Can you extract this logic into a function.

File: src/components/CalendarMonth.jsx [hunk 3]
```diff
@@ -174,13 +212,40 @@ class CalendarMonth extends React.PureComponent {
...
+        if (day && modifiers && modifiers[toISODateString(day)]
+          && (modifiers[toISODateString(day)].has('invalid-span') || modifiers[toISODateString(day)].has('selected-invalid-end'))) {
+          lastInvalidWeek = i;
```
Comment (tofaelahmed) [inline src/components/CalendarMonth.jsx]: It seems you can early break this loop?

File: src/components/CalendarMonth.jsx [hunk 4]
```diff
@@ -174,13 +212,40 @@ class CalendarMonth extends React.PureComponent {
...
+        }
+      });
+    });
+
+    const lastPeriodMonth = endDate && currentMonth && endDate.month() === currentMonth.month();
+
+    const isFirstDay = currentMonth.clone().startOf('month').startOf('isoWeek').isSame(currentMonth.clone().startOf('month'));
+    const firstWeekIndex = currentMonth.clone().startOf('month').startOf('isoWeek').isoWeek() + (isFirstDay ? 0 : 1);
+    const lastWeekIndex = currentMonth.clone().endOf('month').endOf('isoWeek').isoWeek();
+    const activePeriod = `Optimierungszeitraum ${currentMonth.month() + 1} (KW ${firstWeekIndex} - KW ${lastWeekIndex})`;
```
Comment (tofaelahmed) [inline src/components/CalendarMonth.jsx]: hardocded labels (eg: `Optimierungszeitraum`) should not be used

File: src/components/CalendarMonth.jsx [hunk 5]
```diff
@@ -203,41 +268,76 @@ class CalendarMonth extends React.PureComponent {
...
+                    onDayMouseLeave,
+                    onDayClick,
+                    renderDayContents,
+                    phrases,
+                    modifiers: modifiers[toISODateString(day)],
+                    ariaLabelFormat: dayAriaLabelFormat,
+                    currentMonth,
+                    monthIndex,
+                  }))}
+                  <td style={{ padding: '5px 5px 5px 10px', textAlign: 'left' }}>
```
Comment (tofaelahmed) [inline src/components/CalendarMonth.jsx]: to be consistent with the existing coding style please use css in js style

**Review by tofaelahmed — CHANGES_REQUESTED** [review]

File: src/components/DayPickerRangeController.jsx [hunk 6]
```diff
@@ -1061,6 +1094,20 @@ export default class DayPickerRangeController extends React.PureComponent {
...
+  isEndInvalidDate(day) {
+    const { startDate, endDate } = this.props;
+    return isSameDay(day, endDate) && startDate &&
+      (day.month() !== startDate.month() || (day.month() === startDate.month() &&
+      startDate.clone().startOf('isoWeek').month() != startDate.month())) &&
+      (day.clone().startOf('isoWeek').month() == day.month() ||
+        (
+          day.clone().startOf('isoWeek').month() != day.month() &&
+          day.clone().startOf('isoWeek').month() == day.clone().subtract(1,'month').month() &&
+          day.clone().startOf('isoWeek').month() > startDate.month()
```
Comment (tofaelahmed) [inline src/components/DayPickerRangeController.jsx]: you can clone once and reuse.

File: src/components/DayPickerRangeController.jsx [hunk 7]
```diff
@@ -1088,6 +1135,26 @@ export default class DayPickerRangeController extends React.PureComponent {
...
+  isInInvalidSpan(day) {
+    const { startDate, endDate } = this.props;
+    return day.isBetween(startDate, endDate, 'days') && startDate &&
+      (day.month() !== startDate.month() || (day.month() === startDate.month() &&
+      startDate.clone().startOf('isoWeek').month() != startDate.month())) &&
+      (day.clone().startOf('isoWeek').month() == day.month() ||
+        (
+          day.clone().startOf('isoWeek').month() != day.month() &&
+          day.clone().startOf('isoWeek').month() == day.clone().subtract(1,'month').month() &&
+          day.clone().startOf('isoWeek').month() > startDate.month()
```
Comment (tofaelahmed) [inline src/components/DayPickerRangeController.jsx]: same here

File: src/components/CalendarDay.jsx (hunk 1 above)
Comment (mykhailoB8) [reply src/components/CalendarDay.jsx:267]: We are using theme in gastromatic-ui and overriding default values. In gastromatic-react-dates used default values.

File: src/components/CalendarMonth.jsx (hunk 3 above)
Comment (mykhailoB8) [reply src/components/CalendarMonth.jsx]: No. I need to find last item, not first.

File: src/components/CalendarMonth.jsx (hunk 2 above)
Comment (mykhailoB8) [reply src/components/CalendarMonth.jsx]: Done

File: src/components/CalendarMonth.jsx (hunk 4 above)
Comment (mykhailoB8) [reply src/components/CalendarMonth.jsx]: Done

File: src/components/CalendarMonth.jsx (hunk 5 above)
Comment (mykhailoB8) [reply src/components/CalendarMonth.jsx]: Done

File: src/components/DayPickerRangeController.jsx (hunk 6 above)
Comment (mykhailoB8) [reply src/components/DayPickerRangeController.jsx]: Done

File: src/components/DayPickerRangeController.jsx (hunk 7 above)
Comment (mykhailoB8) [reply src/components/DayPickerRangeController.jsx]: Done

**Comment by dehrhard** [comment]
deprecated

**Pull request closed without merging**